in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,270 @@
+
+## L1 Execution Cost
+
+The cost of a block's execution in wei is equal to the following:
+```math
+\begin{aligned}
+\text{execution cost in gas} &= \text{L1 GAS PER BLOCK PROPOSED}  \\
+&+ \text{BLOBS PER BLOCK} * \text{POINT EVALUATION PRECOMPILE GAS} \\
+&+ \frac{\text{L1 GAS PER EPOCH VERIFIED}}{\text{L2 SLOTS PER L2 EPOCH}} \\
+\text{execution cost in wei} &= \text{execution cost in gas} * \text{wei per l1 gas}
+\end{aligned}
+```
+
+## Block Data Cost
+
+The cost of a block's data in wei is equal to the following:
+
+```math
+\text{data cost in wei} = \text{BLOBS PER BLOCK} * \text{GAS PER BLOB} * \text{wei per l1 blob gas}
+```
+
+
+## Exponentially Computed Values 
+
+There are 3 important variables that are computed exponentially:
+- `proving_cost_wei_per_mana`
+- `fee_asset_per_wei`
+- `base_fee_wei_per_mana_congestion_multiplier`
+
+The purpose is to allow prices to fluctuate, but with guarantees, e.g. the proving cost in wei per mana will never change by more than X% per block.
+
+All three computations follow the same formula:
+
+```math
+\text{value} = \text{minimum value} * \text{exp}\left(\frac{\text{variable}}{\text{constant}}\right)
+```
+
+### Example Calculation
+
+[See a calculation](https://www.wolframalpha.com/input?i=%28exp%28%281e9%2B1e9%29%2F%281e9*100%29%29-exp%28%281e9%29%2F%281e9*100%29%29%29%2Fexp%28%281e9%29%2F%281e9*100%29%29) which shows the percent change if the numerator doubles from 1e9 to 2e9 when the denominator is 1e9*100 is ~1%. 
+
+You can repeat the calculation for [a starting point of zero](https://www.wolframalpha.com/input?i=%28exp%28%281*1e9%29%2F%281e9*100%29%29-exp%28%280%29%2F%281e9*100%29%29%29%2Fexp%28%280%29%2F%281e9*100%29%29) to see that the percent change is ~1% as well.
+
+The takeaway is that, in this case, if we cap the change in the numerator between 0 and 1e9, the percent change is at most ~1%.
+
+### `proving_cost_wei_per_mana`
+
+```math
+\text{proving cost wei per mana} = \text{MINIMUM PROVING COST PER MANA} * \text{exp}\left(\frac{\text{proving cost modifier}}{\text{PROVING COST UPDATE FRACTION}}\right)
+```
+
+A proposer can adjust the `proving_cost_modifier` stored in the rollup contract by up to `MAXIMUM_PROVING_COST_PERCENT_CHANGE_PER_BLOCK` each slot.
+
+They do so by updating the `proving_cost_modifier_delta` field, which is used to update the `proving_cost_modifier` stored in the rollup contract.
+
+That is,
+
+```math
+\text{new proving cost modifier} := \text{old proving cost modifier} + \text{proving cost modifier delta}
+```
+
+The new `proving_cost_modifier_delta` is capped at (+/-) `MAXIMUM_PROVING_COST_PERCENT_CHANGE_PER_BLOCK` * `PROVING_COST_UPDATE_FRACTION` / 100.
+
+
+### `fee_asset_per_wei`
+
+```math
+\text{fee asset per wei} = \text{MINIMUM FEE ASSET PRICE} * \text{exp}\left(\frac{\text{fee asset price modifier}}{\text{FEE ASSET PRICE UPDATE FRACTION}}\right)
```

Comment:
How do proposers decrease fee asset per wei? 

i.e. once there is an update that takes `fee asset per wei` above the minimum constant value, what update could the proposers make to decrease it?

Reply:
Ok nvm, this isn't in the form new value = old value * exp... woops